objects-revisited/main.js

```diff
@@ -0,0 +1,123 @@
+const business = {
+  opens: '7am',
+  closes: '8pm',
+  totalEmployees: 4,
+  daysOpen: ['M', 'Tu', 'W', 'Th', 'F'],
+  employees: {
+    marshall: {
+      position: 'c.e.o',
+      daysOfWeekWorking: ['Tu', 'W', 'Th'],
+    },
+    nicholas: {
+      position: 'marketing director',
+      daysOfWeekWorking: ['M', 'W', 'F'],
+    },
+    justin: {
+      position: 'administrative assistant',
+      daysOfWeekWorking: ['M', 'Tu', 'W'],
+    },
+    frankie: {
+      position: 'project manager',
+      daysOfWeekWorking: ['W', 'Th', 'F'],
+    },
+  },
+};
+
+const { daysOpen, employees } = business;
+const employeePositions = [
+  'salesperson',
+  'software engineer',
+  ' c.f.o.',
+  'c.o.o.',
+];
+
+// add weekends to employee schedules
+function addWeekends(object) {
+  const weekends = ['Sa', 'Su'];
+  business.daysOpen = [...daysOpen, ...weekends];
+  for (const employee in object) {
+    object[employee].daysOfWeekWorking =
+      object[employee].daysOfWeekWorking.concat(weekends);
+  }
+}
+
+const createEmployeeObj = {};
+
+// create object of new employees
+function createEmployee(response) {
+  const employee1 = response[0].name.split(' ')[0].toLowerCase();
+  const employee2 = response[1].name.split(' ')[0].toLowerCase();
+  const employee3 = response[2].name.split(' ')[0].toLowerCase();
+  const employee4 = response[3].name.split(' ')[0].toLowerCase();
+  const newEmployees = [employee1, employee2, employee3, employee4];
```

Comment:
Nice, but this would be difficult to update if you decided to do a different number of employees.

One way to deal with this is using `.map`:
```
const newEmployees = response.map((e) => e.name.split(' ')[0].toLowerCase();
```
The problem with that, of course, is that you get more than 4 new employees. You could use `slice` to get just the first 4:
```
const newEmployees = response.slice(0, 4).map((e) => e.name.split(' ')[0].toLowerCase();
```

Another approach is to do this in a for-loop.
```
const newEmployees = [];
for (let i = 0; i < 4; i++) {
  newEmployees.push(response[i].name.split(' ')[0].toLowerCase());
}
```

In both of these solutions, you can easily change the number of employees you process by changing one single number (which you could even pass as a function argument if you wanted).